### PR TITLE
stripe-re-integration

### PR DESCRIPTION
**Pull request by T0MM0R**

### CODE REVIEW

File: src/migrations/m190322_182024_raisers_edge_creds.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function safeDown()
+    {
+        $this->dropTable('{{%raisersedge_creds}}');
+        echo "m190322_182024_raisers_edge_creds cannot be reverted.\n";
```
Comment (dandrust) [inline src/migrations/m190322_182024_raisers_edge_creds.php]: It looks like the table is being dropped on the line above -- is this supposed to be wrapped in a catch block?

File: src/modules/stripereconnector/StripeReConnector.php [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+  /**
+  * @var array
+  */
+  const CHILDSPONSORSHIP_FUNDS = [
+    "BDBA" => "941",
+    "BDBE" => "768",
+    "BDNS" => "388",
+    "BDSA" => "770",
+    "BDSS" => "745",
+    "CORS" => "47",
```
Comment (dandrust) [inline src/modules/stripereconnector/StripeReConnector.php]: Just confirming that this should be two digits as the rest are three.

File: src/modules/stripereconnector/StripeReConnector.php [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+        'accessToken'   => $oauth->getToken(),
+        'refreshToken' => $oauth->getRefreshToken(),
+        'expiresOn' => date("Y-m-d H:i:s", $oauth->getExpires()),
+        'expired' => $oauth->hasExpired()
+      ), ['id' => (int)$accessToken["id"]]
+      )->execute();
+      return $oauth;
+    }
+
+    /**
```
Comment (dandrust) [inline src/modules/stripereconnector/StripeReConnector.php]: Indentation increases from 4 to 5 spaces here

File: src/modules/stripereconnector/StripeReConnector.php [hunk 4]
```diff
@@ -0,0 +1,661 @@
...
+               'Content-Type' => "application/json"
+               ],
+             'body' => $postData
+             ]
+           );
+           return $provider->getParsedResponse($createGiftRequest);
+         }
+
+        /**
+         * @array
```
Comment (dandrust) [inline src/modules/stripereconnector/StripeReConnector.php]: Is this returning an array?  I"m assuming `$provider->getParsedResponse($createAddressRequest)["value"][0]["id"]` is returning a string.

File: src/modules/stripereconnector/StripeReConnector.php (hunk 3 above)
Comment (dandrust) [reply src/modules/stripereconnector/StripeReConnector.php]: and indentation continues to increment for a while with every next function.

File: src/modules/stripereconnector/StripeReConnector.php [hunk 5]
```diff
@@ -0,0 +1,661 @@
...
+              "totalPrice" => $totalPrice,
+              "stripeTransactionId" => $subscriptionId
+            ];
+            $provider = $e->data[0];
+            $accessToken = $e->data[1];
+            $subscriptionKey = $e->data[2];
+
+            // if token is expired get new one.
+            if ( !empty($accessToken) &&
+            time() > strtotime($accessToken["expiresOn"]) ) {
```
Comment (dandrust) [inline src/modules/stripereconnector/StripeReConnector.php]: This condition `!empty($accessToken) && time() > strtotime($accessToken["expiresOn"])` - might consider DRYing it up

File: src/modules/stripereconnector/StripeReConnector.php (hunk 2 above)
Comment (T0MM0R) [reply src/modules/stripereconnector/StripeReConnector.php]: It is 2

File: src/migrations/m190322_182024_raisers_edge_creds.php (hunk 1 above)
Comment (T0MM0R) [reply src/migrations/m190322_182024_raisers_edge_creds.php]: Hmm.  Probably.

File: src/modules/stripereconnector/StripeReConnector.php (hunk 4 above)
Comment (T0MM0R) [reply src/modules/stripereconnector/StripeReConnector.php]: Yes, you're right.  I'll fix that.

File: src/modules/stripereconnector/StripeReConnector.php (hunk 5 above)
Comment (T0MM0R) [reply src/modules/stripereconnector/StripeReConnector.php]: I'll move this inside the refreshToken method.

File: src/modules/stripereconnector/StripeReConnector.php (hunk 3 above)
Comment (T0MM0R) [reply src/modules/stripereconnector/StripeReConnector.php]: Atom was not doing a very nice job with auto indentation.

**Pull request merged**